src/request.ts

```diff
@@ -336,13 +336,17 @@ class DatastoreRequest {
    *   const apiResponse = data[0];
    * });
    */
-  delete(keys, gaxOptions?, callback?) {
-    if (is.fn(gaxOptions)) {
-      callback = gaxOptions;
-      gaxOptions = {};
-    }
-
-    callback = callback || (() => {});
+  delete(keys: Entities): Promise<google.datastore.v1.Datastore.CommitCallback>;
+  delete(keys: Entities, callback: google.datastore.v1.Datastore.CommitCallback): void;
+  delete(
+    keys: Entities, 
+    gaxOptions: CallOptions, 
+    callback: google.datastore.v1.Datastore.CommitCallback): void;
+  delete(keys: Entities, 
+    gaxOptionsOrCallback?: (CallOptions|google.datastore.v1.Datastore.CommitCallback), 
+    cb?: google.datastore.v1.Datastore.CommitCallback): void {
```

Comment:
The return type here should be `void|Promise<google.datastore.v1.Datastore.CommitCallback>`

Reply:
Okay!